File: mgyoutube-multitech/api-python/webservices/VideoWebService.py
import json
from pyramid.response import Response
from pyramid.request import Request
from pyramid.view import view_config
from pyramid.config import Configurator
from pyramid.httpexceptions import HTTPNotFound, HTTPUnauthorized, HTTPMethodNotAllowed,  HTTPServerError, HTTPBadRequest
from webservices.ModuleRepoRegistry import ModuleRepoRegistry


@view_config(route_name='videos-search', request_method='GET')
def videosSearch(request: Request) -> Response:
    searchTerms: str = None
    try:
        searchTerms = request.params.getone("search")
        print("videosSearch: searchTerms='", searchTerms, "'")
    except KeyError:
        print("no 'search' query parameter, failing as 400")
        raise HTTPBadRequest()

    if (searchTerms is None):
        print("search term is None, failing as 400")
        raise HTTPBadRequest()

    searchTerms = searchTerms.strip()
    if (searchTerms == ""):
        print("search term are blank failing as 400")
        raise HTTPBadRequest()

    # // TODO: need to sanitize payload input before using
    sanitizedSearchTerms = searchTerms

    videos = ModuleRepoRegistry.getVideoModule().search(sanitizedSearchTerms)
    print("videosModule.search: videos=", videos)

    responseJson = json.dumps(videos, default=lambda x: x.__dict__)

    print("videosSearch: searchTerms=",
          sanitizedSearchTerms, " returning OK, ", responseJson)
    return Response(body=responseJson, content_type='application/json', charset="UTF-8")


def setupVideoWebService(config: Configurator):
    config.add_route('videos-search', '/api/videos')

File: mgyoutube-multitech/api-node/src/webservices/videowebservice.ts
import { ModuleRepoRegistry } from './modulereporegistry';
import { Request, Response, Server } from 'restify'
import { BadRequestError } from 'restify-errors'
import { UseNext } from './whatsnext'

export class VideoWebService {

    public static setupRouter(server: Server) {
        // DRP: restify can't seem to use class methods for handlers here... they lose the "this" pointer reference
        server.get('/api/videos', async (req, res, next) => {
            return UseNext.handleAsyncRestifyCall(await VideoWebService.searchVideos(req, res), next);
        });
    }

    private static async searchVideos(req: Request, res: Response): Promise<UseNext> {

        var searchTerms: string = req.query["search"]
        console.log("videosSearch: searchTerms='" + searchTerms + "'");

        if (!searchTerms) {
            console.warn("search term are blank failing as 400");
            const err400 = new BadRequestError();
            return new UseNext(err400);
        }
        else {
            // TODO: need to sanitize payload input before using
            var sanitizedSearchTerms = searchTerms;

            var videos = await ModuleRepoRegistry.getVideosModule().search(searchTerms);

            console.log("videosSearch: searchTerms=" + sanitizedSearchTerms + " returning OK, " + videos);
            res.setHeader("content-type", "application/json")
            res.send(videos)

            return UseNext.Nothing;
        }
    }
}